google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -210,8 +209,8 @@ def _client_session(self) -> aiohttp.ClientSession:
     _project: str
     _region: str
 
-    _current: Union[Coroutine, asyncio.Task]
-    _next: Union[Coroutine, asyncio.Task]
+    _current: Union[concurrent.futures.Future, asyncio.Task]
```

Comment:
I thought we weren't using futures any more?

Reply:
So the initial perform_refresh has to be kicked off from another thread using `run_coroutine_threadsafe`. When I had current set to a coroutine and tried to await it, I got an error that said `RuntimeError: cannot reuse already awaited coroutine`. If we use a wrapped future just for that initial request, we can call `await` on it as many times as we want, and if it's already completed, we just get the result. I just updated the code again to store the perform_refresh coroutine again, but I think using a future just for that first request is a better approach

Reply:
Setting it to a coroutine gives this error: https://source.cloud.google.com/results/invocations/43cbe38e-181c-48d1-ba35-bea135868b7c/targets/cloud-devrel%2Fcloud-sql%2Fpython-connector%2Fpython3.7%2Flinux%2Fpresubmit/log

Just set it back to a future.